Source/WebKit/NetworkProcess/storage/CacheStorageDiskStore.cpp

```diff
@@ -309,21 +327,18 @@ std::optional<CacheStorageRecord> CacheStorageDiskStore::readRecordFromFileData(
     if (!responseBody)
         return std::nullopt;
 
-    CacheStorageRecordInformation info { metaData->key, header->insertionTime, 0, 0, header->responseBodySize, header->request.url(), false, { } };
-    info.updateVaryHeaders(header->request, header->response.httpHeaderField(WebCore::HTTPHeaderName::Vary));
-    return CacheStorageRecord { info, header->requestHeadersGuard, header->request, header->options, header->referrer, header->responseHeadersGuard, header->response.crossThreadData(), header->responseBodySize, WTFMove(*responseBody) };
+    return CacheStorageRecord { storedInfo->info, storedInfo->header.requestHeadersGuard, storedInfo->header.request, storedInfo->header.options, storedInfo->header.referrer, storedInfo->header.responseHeadersGuard, storedInfo->header.response.crossThreadData(), storedInfo->header.responseBodySize, WTFMove(*responseBody) };
 }
 
-void CacheStorageDiskStore::readAllRecords(ReadAllRecordsCallback&& callback)
+void CacheStorageDiskStore::readAllRecordInfos(ReadAllRecordInfosCallback&& callback)
 {
-    auto didReadRecordFiles = [this, protectedThis = Ref { *this }, callback = WTFMove(callback)](auto fileDatas, auto blobDatas) mutable {
-        ASSERT(fileDatas.size() == blobDatas.size());
-        Vector<CacheStorageRecord> result;
+    auto didReadRecordFiles = [this, protectedThis = Ref { *this }, callback = WTFMove(callback)](auto fileDatas) mutable {
+        Vector<CacheStorageRecordInformation> result;
         for (size_t index = 0; index < fileDatas.size(); ++index) {
-            if (auto record = readRecordFromFileData(fileDatas[index], blobDatas[index]))
-                result.append(WTFMove(*record));
+            if (auto storedInfo = readRecordInfoFromFileData(m_salt, fileDatas[index]))
+                result.append(WTFMove(storedInfo->info));
```

Comment:
We can probably use WTF::map here to do the reserve/uncheckedAppend thing.

Reply:
hmm checking the code, `readRecordInfoFromFileData` could fail, so I am not sure we could WTF::map here. I will use reserve/uncheckedAppend explicitly instead.